opentelemetry-otlp/src/metric.rs

```diff
@@ -113,7 +131,10 @@ where
     }
 
     /// Build with the aggregator selector
-    pub fn with_aggregator_selector(self, aggregator_selector: AS) -> Self {
+    pub fn with_aggregator_selector(
```

Comment:
Perhaps a simpler way of fixing this to allow users to correctly specify their own aggregator selector impls would be to change this method to be something along the lines of:

```rust
/// Build with the aggregator selector
pub fn with_aggregator_selector<T>(
    self,
    aggregator_selector: T,
) -> OtlpMetricPipelineBuilder<T, ES, SP, SO, I, IO>
where
    T: AggregatorSelector + Send + Sync + 'static,
{
    OtlpMetricPipelineBuilder {
        aggregator_selector,
        export_selector: self.export_selector,
        spawn: self.spawn,
        interval: self.interval,
        export_config: self.export_config,
        tonic_config: self.tonic_config,
        resource: self.resource,
        stateful: self.stateful,
        period: self.period,
        timeout: self.timeout,
    }
}
```

That way users can pass the type directly instead of having to manually box

Reply:
cool, done